### PR TITLE
ffmpeg: allow msvc shared with conan v1 & ffmeg < 6.1.1

### PR DESCRIPTION
**Pull request by SpaceIm**
### Summary
Changes to recipe:  **ffmpeg/all**

#### Motivation

Allow ffmpeg shared again with msvc & conan v1 when version < 6.1.1 (regression from https://github.com/conan-io/conan-center-index/pull/22195)

#### Details

Build of ffmpeg shared with msvc & conan v1 and version < 6.1.1 was working fine, and actually in https://github.com/conan-io/conan-center-index/pull/22195 there was no issue in CI, just for 6.1.1, so no reason to prevent shared build for all versions.

Moreover short_paths was an attempt for fix this in this PR, but there was no improvement when you look at CI logs, so no reason to keep it.

---
- [ ] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [ ] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [ ] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — DISMISSED** [review]
Thanks! We should have realized that old versions were still ok to build as shared, thanks for taking the time to revert it for those versions :)

**Comment by AbrilRBS** [comment]
The build issues seem to be related to what @uilianries and I are looking into for this recipe, will update you once that's addressed :)

**Comment by Drllap** [comment]
@SpaceIm @AbrilRBS @uilianries What is the status here? Do I understand it correctly that the shared lib version on msvc only fails on v6.1.1 and v6.1. If I create a MR that disables shared builds for only those two could that me merged?

**Comment by uilianries** [comment]
I rebased the branch over master. Locally, I can build using ffmpeg:shared=True for 6.1 (Conan 1.x)

File: recipes/ffmpeg/all/conanfile.py [hunk 1]
```diff
@@ -345,7 +343,7 @@ def validate(self):
                 raise ConanInvalidConfiguration("FFmpeg '{}' option requires '{}' option to be enabled".format(
                     dependency, "' or '".join(features)))
 
-        if conan_version.major == 1 and is_msvc(self) and self.options.shared:
+        if Version(self.version) >= "6.1.1" and conan_version.major == 1 and is_msvc(self) and self.options.shared:
```
Comment (uilianries) [inline recipes/ffmpeg/all/conanfile.py:346]:
```suggestion
        if Version(self.version) >= "6.1" and conan_version.major == 1 and is_msvc(self) and self.options.shared:
```

**Review by uilianries — APPROVED** [review]
LGTM.

**Pull request merged**